File: day1/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
)

func spelledToInt(spelled string) int {
	m := map[string]int{
		"zero":  0,
		"one":   1,
		"two":   2,
		"three": 3,
		"four":  4,
		"five":  5,
		"six":   6,
		"seven": 7,
		"eight": 8,
		"nine":  9,
	}

	output, ok := m[spelled]
	if !ok {
		output, _ := strconv.Atoi(spelled)
		// fmt.Println("input:", spelled, "output:", (output))
		return output
	}
	// fmt.Println("input:", spelled, "output:", (output))
	return output

}

func reverseString(input string) string {
	output := ""
	for _, c := range input {
		output = string(c) + output
	}
	return output
}

func main() {
	//part 1
	readFile, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	fileScanner := bufio.NewScanner(readFile)
	fileScanner.Split(bufio.ScanLines)

	output := 0
	re_string := "\\d"
	r, _ := regexp.Compile(re_string)

	for fileScanner.Scan() {
		line := fileScanner.Text()
		// fmt.Println(line)

		match := r.FindString(line)
		// fmt.Println(match)

		match_b := r.FindString(reverseString(line))
		// fmt.Println(match_b)

		line_value, _ := strconv.Atoi(match + match_b)
		// fmt.Println(line_value)

		output += line_value
	}

	fmt.Println(output)

	readFile.Close()

	//part 2
	readFile, err = os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	fileScanner = bufio.NewScanner(readFile)
	fileScanner.Split(bufio.ScanLines)

	output = 0
	re_string = "(?:zero|one|two|three|four|five|six|seven|eight|nine|\\d)"
	temp_b := reverseString("zero|one|two|three|four|five|six|seven|eight|nine")
	re_string_b := fmt.Sprintf("(?:%s|\\d)", temp_b)
	r, _ = regexp.Compile(re_string)
	r_b, _ := regexp.Compile(re_string_b)

	for fileScanner.Scan() {
		line := fileScanner.Text()
		match := r.FindString(line)
		match_b := r_b.FindString(reverseString(line))
		match_b = reverseString(match_b)

		// fmt.Println(line)
		// fmt.Print(match)
		// fmt.Println(spelledToInt(match))
		// fmt.Print(match_b)
		// fmt.Println(spelledToInt(match_b))

		num := strconv.Itoa(spelledToInt(match))
		num_b := strconv.Itoa(spelledToInt(match_b))
		// fmt.Println(num, num_b)
		line_value, _ := strconv.Atoi(num + num_b)
		// fmt.Println(line_value)

		output += line_value
	}

	fmt.Println(output)

	readFile.Close()
}

File: day1/day1.py
import re

def spelled_out_to_int(spelled_number):
    number_dict = {
        'zero': 0, 'one': 1, 'two': 2, 'three': 3, 'four': 4,
        'five': 5, 'six': 6, 'seven': 7, 'eight': 8, 'nine': 9
    }

    # Convert the spelled-out number to lowercase for case-insensitive matching
    spelled_number_lower = spelled_number.lower()

    # Check if the spelled-out number is in the dictionary
    if spelled_number_lower in number_dict:
        return number_dict[spelled_number_lower]
    else:
        return spelled_number

with open("input.txt") as f:
    day1_in = f.read()
    output = 0
    for line in day1_in.splitlines():
        match = re.search(r'\d', line)
        match2 = re.search(r'\d', line[::-1])
        output += int(f'{match.group()}{match2.group()}') 

    # print(output)

#part2


regex_new ='(?:zero|one|two|three|four|five|six|seven|eight|nine|\d)' 
pattern_rev = 'zero|one|two|three|four|five|six|seven|eight|nine'[::-1]
regex_new_b =f'(?:{pattern_rev}|\d)' 
# print(pattern_rev, regex_new_b)

with open("input.txt") as f:
    day1_in = f.read()
    output = 0
    count = 0
    for line in day1_in.splitlines():
        match_f = re.search(regex_new,line)
        match_b = re.search(regex_new_b,line[::-1])

        # print(line,'\n',match_f,match_b)
        match_f_int = spelled_out_to_int(match_f.group())
        match_b_int = spelled_out_to_int(match_b.group()[::-1])
        # print(match_f_int, match_b_int)
        output += int(f'{match_f_int}{match_b_int}')

    print(output)